archivebox/core/migrations/0007_archiveresult.py

```diff
@@ -0,0 +1,91 @@
+# Generated by Django 3.0.8 on 2020-11-04 12:25
+
+import json
+from pathlib import Path
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+from config import CONFIG
+from index.json import to_json
+
+
+def forwards_func(apps, schema_editor):
+    from core.models import EXTRACTORS
+
+    Snapshot = apps.get_model("core", "Snapshot")
+    ArchiveResult = apps.get_model("core", "ArchiveResult")
+
+    snapshots = Snapshot.objects.all()
+    for snapshot in snapshots:
+        out_dir = Path(CONFIG['ARCHIVE_DIR']) / snapshot.timestamp
+
+        try:
+            with open(out_dir / "index.json", "r") as f:
+                fs_index = json.load(f)
+        except Exception as e:
+            continue
+
+        history = fs_index["history"]
+
+        for extractor in history:
+            for result in history[extractor]:
+                ArchiveResult.objects.create(extractor=extractor, snapshot=snapshot, cmd=json.dumps(result["cmd"]), cmd_version=result["cmd_version"], 
```

Comment:
`cmd=json.dumps(result["cmd"])` why json.dumps here instead of `' '.join()`? should we make it an ArrayField instead so it can hold multiple args natively without string splitting/quoting problems?

Reply:
I used a json just to keep it as similar to the original ArchiveResult as possible. It is originally a list, and while I could use `join` and `split`, I think that json is more clear on the intent. Also, if there is something like `cmd=["some random", "command"]` the join-split will generate a cmd that is different than the original.

Reply:
Why not use `ArrayField` or `JSONField`?

Reply:
New in Django 3.1. I will try to bump the django version.